### PR TITLE
Added content to Javascript This keyword

### PR DESCRIPTION
**Pull request by Ryrden**
-  Added content to javascript this keyword - #1838 
- blank spaces removed in existing files

### CODE REVIEW

**Comment by Ryrden** [comment]
Hii @kamranahmedse, I have added the content in 111 - javascript this keyword

**Comment by Ryrden** [comment]
With the reference of the issue #1838 . I've added the following contents in the Javascript Roadmap

Contents for : 
- Function Borrowing (No conflict)
- Explicit Binding
- In a method 
- In a Function
- Using it alone
- In event handlers
- In arrow functions

@kamranahmedse Please review this PR (Pull Request) and label this PR as ***"hacktoberfest-accepted"***.

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
-# Function Borrowing
+# Function borrowing
 
-Function borrowing allows us to use the methods of one object on a different object without having to make a copy of that method and maintain it in two separate places. It is accomplished through the use of `.call()`, `.apply()`, or `.bind()`, all of which exist to explicitly set this on the method we are borrowing.
+**_Method borrowing_**, also known as function borrowing, is a technique where a method of one object is used on another object. This is possible because JavaScript functions are objects and can be passed around like any other object.
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md:3]:
```suggestion
Function borrowing allows us to use the methods of one object on a different object without having to make a copy of that method and maintain it in two separate places. It is accomplished through the use of `.call()`, `.apply()`, or `.bind()`, all of which exist to explicitly set this on the method we are borrowing.
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
-# Function Borrowing
+# Function borrowing
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md:1]:
```suggestion
# Function Borrowing
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md [hunk 3]
```diff
@@ -2,4 +2,8 @@
 
 Function borrowing allows us to use the methods of one object on a different object without having to make a copy of that method and maintain it in two separate places. It is accomplished through the use of `.call()`, `.apply()`, or `.bind()`, all of which exist to explicitly set this on the method we are borrowing.
 
-<BadgeLink colorScheme="yellow" badgeText="Read" href="https://medium.com/@ensallee/function-borrowing-in-javascript-4bd671e9d7b4">Function borrowing</BadgeLink>
+In JavaScript, we can reuse the method of a function on a different object other than the object it was defined on. Method borrowing helps to keep us from having to write the same code multiple times. Using the predefined JavaScript methods, `call()`, `apply()` or `bind()`, we can borrow methods from other objects without inheriting their properties and methods.
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.educative.io/answers/what-is-method-borrowing'>Method Borrowing</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/100-function-borrowing.md:8]:
```suggestion
<ResourceGroupTitle>Free Content</ResourceGroupTitle>
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/102-this-in-a-function.md [hunk 4]
```diff
@@ -1 +1,23 @@
...
+
+**Output**
+
+```js
+Hello, my name is undefined
+```
+
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this in a function</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/102-this-in-a-function.md:23]:
```suggestion
# this in a function

The keyword `this` when used in a function refers to the global object.

*Note: in a browser window the global object is the `window` object.*

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this in a function</BadgeLink>
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/103-using-this-alone.md [hunk 5]
```diff
@@ -1 +1,21 @@
...
+```
+
+**Output**
+
+```js
+Window {parent: Window, opener: null, top: Window, length: 0, frames: Window, …}
+```
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this Alone</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/103-using-this-alone.md:21]:
```suggestion
# Using this alone

The keyword `this` when used alone refers to the global object.

*Note: in a browser window the global object is the `window` object.*

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this Alone</BadgeLink>
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/104-this-in-event-handlers.md [hunk 6]
```diff
@@ -1 +1,21 @@
...
+```
+
+**Output**
+
+```js
+<button>Click me</button>
+```
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this in Event handlers</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/104-this-in-event-handlers.md:21]:
```suggestion
# this in event handlers

The keyword `this` when used in an event handler refers to the element that received the event.

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='yellow' badgeText='Read' href='https://www.w3schools.com/js/js_this.asp'>this in Event handlers</BadgeLink>
```

File: content/roadmaps/106-javascript/content/111-javascript-this-keyword/105-this-in-arrow-functions.md [hunk 7]
```diff
@@ -1 +1,24 @@
...
+```
+
+**Output**
+
+```js
+Hello, my name is undefined
+```
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='yellow' badgeText='Read' href='https://stackoverflow.com/questions/66518020/javascript-this-keyword-and-arrow-function'>this keyword and arrow function</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/106-javascript/content/111-javascript-this-keyword/105-this-in-arrow-functions.md:24]:
```suggestion
# this in arrow functions

The keyword `this` when used in an arrow function refers to the parent object.

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='yellow' badgeText='Read' href='https://stackoverflow.com/questions/66518020/javascript-this-keyword-and-arrow-function'>this keyword and arrow function</BadgeLink>
```

**Pull request merged**